src/ITILSolution.php

```diff
@@ -80,9 +80,9 @@ public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
 
     public static function canView()
     {
-        return Session::haveRight('ticket', READ)
-             || Session::haveRight('change', READ)
-             || Session::haveRight('problem', READ);
+        return Session::haveRightsOr('ticket', [Ticket::READMY, Ticket::READALL, Ticket::READGROUP, Ticket::READASSIGN])
+            || Session::haveRightsOr('change', [Change::READMY, Change::READALL])
+            || Session::haveRightsOr('problem', [Problem::READMY, Problem::READALL]);
```

Comment:
I am not sure of my proposal, but I guess it would prevent code duplication without introducing a lack of checks.

```suggestion
        return Ticket::canView() || Problem::canView() || Change::canView();
```

Reply:
It doesn't change anything for Change and Problem, but for Tickets, it also checks the validation. I don't know if that's really relevant?

![image](https://github.com/user-attachments/assets/0d7d90bf-a496-4354-8f08-229e3c457372)

Reply:
If you need to validate a ticket, then it is probably better to be able to see any refused solution that is present in this ticket. I guess it will not result in permission issues, as, if you cannot see a precise ticket, you will be blocked by the `canViewItem()` check that will anyway prevent you to see any of its solutions.